packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-swift-visitor.test.ts.snap

```diff
@@ -0,0 +1,285 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`AppSyncSwiftVisitor Should handle nullability of lists appropriately 1`] = `
+"// swiftlint:disable all
+import Amplify
+import Foundation
+
+public struct ListContainer: Model {
+  public let id: String
+  public var name: String?
+  public var list: [Int?]?
+  public var requiredList: [String?]
+  public var requiredListOfRequired: [StatusEnum]
+  public var listOfRequired: [Bool]?
+  public var requiredListOfRequiredDates: [Temporal.Date]
+  public var listOfRequiredFloats: [Double]?
+  public var requiredListOfCustomTypes: [CustomType?]
+  public var createdAt: Temporal.DateTime?
+  public var updatedAt: Temporal.DateTime?
+  
+  public init(id: String = UUID().uuidString,
+      name: String? = nil,
+      list: [Int?]? = nil,
+      requiredList: [String?] = [],
+      requiredListOfRequired: [StatusEnum] = [],
+      listOfRequired: [Bool]? = nil,
+      requiredListOfRequiredDates: [Temporal.Date] = [],
+      listOfRequiredFloats: [Double]? = nil,
+      requiredListOfCustomTypes: [CustomType?] = []) {
+    self.init(id: id,
+      name: name,
+      list: list,
+      requiredList: requiredList,
+      requiredListOfRequired: requiredListOfRequired,
+      listOfRequired: listOfRequired,
+      requiredListOfRequiredDates: requiredListOfRequiredDates,
+      listOfRequiredFloats: listOfRequiredFloats,
+      requiredListOfCustomTypes: requiredListOfCustomTypes,
+      createdAt: nil,
+      updatedAt: nil)
+  }
+  internal init(id: String = UUID().uuidString,
+      name: String? = nil,
+      list: [Int?]? = nil,
+      requiredList: [String?] = [],
+      requiredListOfRequired: [StatusEnum] = [],
+      listOfRequired: [Bool]? = nil,
+      requiredListOfRequiredDates: [Temporal.Date] = [],
+      listOfRequiredFloats: [Double]? = nil,
+      requiredListOfCustomTypes: [CustomType?] = [],
+      createdAt: Temporal.DateTime? = nil,
+      updatedAt: Temporal.DateTime? = nil) {
+      self.id = id
+      self.name = name
+      self.list = list
+      self.requiredList = requiredList
+      self.requiredListOfRequired = requiredListOfRequired
+      self.listOfRequired = listOfRequired
+      self.requiredListOfRequiredDates = requiredListOfRequiredDates
+      self.listOfRequiredFloats = listOfRequiredFloats
+      self.requiredListOfCustomTypes = requiredListOfCustomTypes
+      self.createdAt = createdAt
+      self.updatedAt = updatedAt
+  }
+}"
```

Comment:
how come there is an internal init here?

Reply:
@diegocstn when timestamp is enabled, then there is an internal init? when it is not, then there is no internal init? what does timestamp enablement mean?

Reply:
@lawmicha `timestampEnabled` is a feature flag to support `createdAt` and `updatedAt` fields, It's enabled by default on new projects. The internal initializer is only used internally to decode model instances with read-only fields. The public initializer doesn't accept `createdAt` and `updatedAt` parameters